### DISCUSSION TITLE
Automatic Redirect with StreamProvider (instead of ChangeNotifier)

### DISCUSSION THREAD

**Discussion by kornha** [csells/go_router#181] [Q&A]
Hello,

Great library, thank you so much!

I wanted to ask a question that I think would help a lot of firebase users. It can become simpler in firebase to use a StreamProvider instead of a ChangeNotifierProvider because 
`_auth.authStateChanges();` returns a stream, thereby one can simply do thus:

```
  Stream<User?> get fbUser {
    return _auth.authStateChanges();
  }
```
However, I cannot quite figure how I might use this type of provider with redirect and refreshListenable. (Obviously I could abandon the StreamProvider, or intercept with a ChangeNotifier, but not ideal). Any thoughts?

**Comment by csells** [comment]
It would be easy to translate a stream of events into change notifications for a listenable implementation.

**Reply by csells** [reply]
in fact, this may help: https://github.com/csells/go_router/discussions/122#discussioncomment-1587429
